eos/form-factors/parametric-kkrvd2024.cc

```diff
@@ -0,0 +1,371 @@
+    KKRvD2024FormFactors<VacuumToPiPi>::f_p(const complex<double> & q2) const
+    {
+        const auto z        = this->z(q2);
+        const auto chi      = 0.0258815; // GeV^-2, from Meril at Q^2 = 1 GeV^2
+        const auto phitilde = this->phitilde_p(z, chi);
+
+        // Super-threshold pole location
+        const auto zr = this->_zr(this->_M_fp_I1(), this->_G_fp_I1());
+
+        // prepare expansion coefficients
+        std::array<double, 10> b;
+        std::copy(_b_fp_I1.cbegin(), _b_fp_I1.cend(), b.begin()+2);
+        // Fix b[0] and b[1] to enforce F(q2=0) = 1 and F'(q2=t+) = 0
+        b[0] = _b0_fp_I1(chi, zr);
+        b[1] = _b1_fp_I1(chi, zr);
+
+        const auto series = this->series_m(z, b);
+
+        return series / (z - zr) / (z - std::conj(zr)) /  phitilde;
+    }
+
+    complex<double>
+    KKRvD2024FormFactors<VacuumToPiPi>::f_t(const double & /*q2*/) const
+    {
+        throw InternalError("Not implemented!");
+        return 0.0;
+    }
+
+    complex<double>
+    KKRvD2024FormFactors<VacuumToPiPi>::f_0(const double & /*q2*/) const
+    {
+        return 0.0; // vanishes in our approximation
+    }
+
+    double
+    KKRvD2024FormFactors<VacuumToPiPi>::b_0() const
+    {
+        const auto chi = 0.0258815; // GeV^-2, from Meril at Q^2 = 1 GeV^2
+        const auto zr  = this->_zr(this->_M_fp_I1(), this->_G_fp_I1());
+
+        return _b0_fp_I1(chi, zr);
+    }
+
+    double
+    KKRvD2024FormFactors<VacuumToPiPi>::b_1() const
+    {
+        const auto chi = 0.0258815; // GeV^-2, from Meril at Q^2 = 1 GeV^2
+        const auto zr  = this->_zr(this->_M_fp_I1(), this->_G_fp_I1());
+
+        return _b1_fp_I1(chi, zr);
+    }
+
+    double
+    KKRvD2024FormFactors<VacuumToPiPi>::dFdq2_q2eq0() const
+    {
+        const double z0 = std::real(this->z(0.0));
+        const double chi = 0.0258815; // GeV^-2, from Meril at Q^2 = 1 GeV^2
+
+        const double phitilde_z0      = std::real(this->phitilde_p(z0, chi));
+        const double phitildeprime_z0 = std::real(this->phitildeprime_p(z0, chi));
+
+        // Super-threshold pole location
+        const auto zr =  this->_zr(this->_M_fp_I1(), this->_G_fp_I1());
+
+        // prepare expansion coefficients
+        std::array<double, 10> b;
+        std::copy(_b_fp_I1.cbegin(), _b_fp_I1.cend(), b.begin()+2);
+        // Fix b[0] and b[1] to enforce F(q2=0) = 1 and F'(q2=t+) = 0
+        b[0] = _b0_fp_I1(chi, zr);
+        b[1] = _b1_fp_I1(chi, zr);
+
+        const double sum1 = std::real(series_m(z0, b));
+        double sum2 = 0.0;
+        for (auto i = 0u; i < b.size(); i++)
+        {
+            sum2 += b[i] * i * std::pow(z0, i-1);
+        };
+
+        const double xprime_z0 = ( 2.0 * (std::real(zr) - z0) * phitilde_z0 - std::norm(z0 - zr) * phitildeprime_z0 ) / power_of<2>(std::norm(z0 - zr) * phitilde_z0);
+
+        const double dFdz_z0 = sum1 * xprime_z0 + sum2 / (phitilde_z0 * std::norm(z0 - zr)); // the weight factor has been absorbed into 1 / phi
```

Comment:
The comment is either incorrect or indicates a bug: the weight factor should not be completely absorbed into 1/phi, since we now have $F \sim (1 - z)^2$, right?

Ah, you did absorb it. Not sure that I like this, but it seems to work.

Reply:
I'm just going to remove the comment - now everything is called phitilde and the comment for that function tesll you what has been done